pkg/util/featuregates.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	upstreamfeature "k8s.io/component-base/featuregate"
+	cloudfeatures "k8s.io/controller-manager/pkg/features"
+)
+
+// GetEnabledDisabledFeatures returns two slices that contain all the known feature gates
+// and separates them by their enabled/disabled state. It filters features so that we return
+// only those that are to be consumed by the cloud provider instead of all that exist
+// in the kubernetes ecosystem.
+func GetEnabledDisabledFeatures(features featuregates.FeatureGate) ([]string, []string) {
```

Comment:
Because you're now using this in main, it will filter all the CCM feature gates to be only those upstream. This prevents us using any feature gates that are CCMO specific. I think the filtering logic needs to be moved close to where we inject the feature gates into the CCM configuration

Reply:
There are two points where feature gates are used. Firstly, they are logged in the main.go file, here, the feature gates should be the full openshift set. The second is injected into the assets for the different providers. These second ones are the ones that need to be filtered out.
The CCMO itself doesn't care about extra feature gates which means it doesn't need them filtered early like you currently have it

Reply:
Yeah that's right. So my concern is still valid (i.e. what to do with custom-developed feature gates for cloud provider), but as we don't have any instance today (the only flag that we need to pass is upstream one) I defer it to be a problem of future us (hopefully never)